tools/continuous_integration/bisect.py

```diff
@@ -0,0 +1,104 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A script to bisect the mainline commits and find the culprit commit.
```

Comment:
I am trying to understand the difference between this tool and `git bisect`.  According to this comment, it seems that this tool checks only commits on the $BRANCH branch?  Also, When I read the code, it seems that this tool also calls ctest repeatedly -- is that for debug unit tests with random behavior? Would it be easier for users to understand about this tool if we have a Github issue listing the design purposes?

Reply:
Yes, it only check BRANCH branch. (mainline). git bisect also checks feature branch. I found commits in Paddle's feature branch are not guaranteed to pass tests or compile, so it bisect there won't work anyway.

Yes, the repeat is used to debug flaky behavior. I'll create a issue.

Reply:
https://github.com/PaddlePaddle/Paddle/issues/8449